moto/iam/models.py

```diff
@@ -633,7 +646,7 @@ def delete_policy(self, policy_name):
 
     @property
     def physical_resource_id(self):
-        return self.id
+        return self.name
```

Comment:
This was an (unrelated) bug in the implementation. Verified it against AWS itself - the physical resource id is always the role name